src/components/App.js

```diff
@@ -488,100 +484,102 @@ export default class App extends Component {
     })
   }
 
-  onCloseUsernameModal(){
+  onCloseUsernameModal() {
     this.setState({
       showUsernameModal: false
     });
   }
 
-  onSetUsername(username){
-    if (!username){
+  onSetUsername(username) {
+    if (!username) {
       this.onError('Invalid username!');
       return;
     }
     localStorage.setItem(USERNAME_KEY, username);
+
+    this.props.setUsername(username);
+
     this.setState({
-      username: username,
       status: 'viewing'
     });
 
     this.onCloseUsernameModal();
   }
 
-  onSendMessage(message){
+  onSendMessage(message) {
     this.createMessage(message);
   }
 
-  sendJsonMessage(contents, tags, ttl_secs=0){
+  sendJsonMessage(contents, tags, ttl_secs = 0) {
     console.log("Creating message with contents", contents);
 
     let saveName = tags.join('|||');
     let fileBlob = new Blob([JSON.stringify(contents)],
-                            {type: 'application/json'})
+      { type: 'application/json' })
     fileBlob.name = saveName;
 
     console.log("Encrypting file blob");
 
     let mID = this.state.mID;
-    if (this.state.keyPairReady){
+    if (this.state.keyPairReady) {
       miniLock.crypto.encryptFile(fileBlob, saveName, [mID],
-                                  mID, this.state.keyPair.secretKey,
-                                  this.sendMessageToServer.bind(this, ttl_secs));
+        mID, this.state.keyPair.secretKey,
+        this.sendMessageToServer.bind(this, ttl_secs));
     } else {
       let interval = setInterval(() => {
-        if (!this.state.keyPairReady){
+        if (!this.state.keyPairReady) {
           return;
         }
 
         miniLock.crypto.encryptFile(fileBlob, saveName, [mID],
-                                    mID, this.state.keyPair.secretKey,
-                                    this.sendMessageToServer.bind(this, ttl_secs));
+          mID, this.state.keyPair.secretKey,
+          this.sendMessageToServer.bind(this, ttl_secs));
         clearInterval(interval);
       }, 500)
     }
   }
 
-  sendStatusMessage(){
+  sendStatusMessage() {
     // This is in a setInterval because sometimes `this.state.keyPair`
     // isn't quite ready yet
     let interval = setInterval(() => {
       let username = this.state.username;
       let status = this.state.status;
 
-      if (!username || !status || !this.state.keyPairReady){
+      if (!username || !status || !this.state.keyPairReady) {
         return;
       }
 
       let contents = {};  // Unused. TODO: Make more efficient?
-      let tags = ['from:'+username, 'type:userstatus', 'status:'+status];
+      let tags = ['from:' + username, 'type:userstatus', 'status:' + status];
```

Comment:
Hmm, weird. There are no .editorconfig or .eslint for the project. Though there should probably be one. I am using Visual Studio Code with some plugins and settings for this but i had no idea that it took the liberty without explicit configurations in the codebase.